### PR TITLE
Added Oracle Object Storage adapter

### PR DESCRIPTION
**Pull request by SaketKaswa20**
Added Oracle Object Storage Adapter 

Solving Issue #6415

### CODE REVIEW

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Great PR! 🤯 We left some comments during the review, please check them out.

Please also make sure to:

1. to run the formatter to fix the lint errors
2. attach a screenshot of successful tests

File: src/Storage/Device/OracleObject.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    /**
+     * OracleObject constructor.
+     *
+     * @param string $root
+     */
+    public function __construct(string $root = '')
+    {
+        $this->root = $root;
+    }
```
[the remark this answers is not in the document]
Comment (SaketKaswa20) [reply src/Storage/Device/OracleObject.php:24]: Thanks!
Will solve the mentioned things and push the changes.

**Comment by SaketKaswa20** [comment]
I am facing problem in phpunit test. It is showing could not read phpunit.xml when using `phpunit --configuration phpunit.xml` and when running `vendor/bin/phpunit --configuration phpunit.xml` I get the error zsh: no such file or directory: vendor/bin/phpunit.

**Comment by gewenyu99** [comment]
@stnguyen90 take a peek at the comment here

**Comment by stnguyen90** [comment]
Discord thread: https://discord.com/channels/564160730845151244/1162704045937541140

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Changes still pending.

Discord thread: https://discord.com/channels/564160730845151244/1162704045937541140

File: src/Storage/Device/OracleObject.php [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    /**
+     * OracleObject constructor.
+     *
+     * @param string $root
+     * @param array $config Configuration options for Oracle Object Storage
+     */
+    public function __construct(string $root = '', array $config = [])
+    {
+        $this->root = $root;
+        $this->config = $config;
```
Comment (stnguyen90) [inline src/Storage/Device/OracleObject.php:25]: Let's not use a config array like this. Please take a look at how similar adapters are implemented in this repo to see how the configuration should be implemented.

File: src/Storage/Device/OracleObject.php [hunk 3]
```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace Utopia\Storage\Device;
+
+use Exception;
+use Utopia\Storage\Device;
+use Utopia\Storage\Storage;
+
+class OracleObject extends Device
```
Comment (stnguyen90) [inline src/Storage/Device/OracleObject.php:9]: This should probably extend the [S3](https://github.com/utopia-php/storage/blob/main/src/Storage/Device/S3.php) adapter because Oracle Object is an S3-compatible storage provider.

File: src/Storage/Device/OracleObject.php [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        return true; // Example: always return true for the sake of testing
+    }
+
+    public function deleteFileFromOracleObjectStorage(string $remoteFilePath): bool
+    {
+        // Implement logic to delete a file from Oracle Object Storage
+        // Return true on success, false on failure
+        // You should use the Oracle Object Storage SDK or API for this operation
+        return true; // Example: always return true for the sake of testing
+    }
```
Comment (stnguyen90) [inline src/Storage/Device/OracleObject.php:177]: These custom functions should be added because this should implement [Device](https://github.com/utopia-php/storage/blob/main/src/Storage/Device.php) such that you would use the same methods from the Device class (like [upload](https://github.com/utopia-php/storage/blob/main/src/Storage/Device.php#L102) and [delete](https://github.com/utopia-php/storage/blob/983e6dee137012f9f57f126d3c79aab54e4e8824/src/Storage/Device.php#L192)).

File: src/Storage/Storage.php [hunk 5]
```diff
@@ -21,6 +21,8 @@ class Storage
 
     const DEVICE_LINODE = 'linode';
 
+    const DEVICE_ORACLEOBJECT = 'oracleobject';
```
Comment (stnguyen90) [inline src/Storage/Storage.php:24]: This should be `DEVICE_ORACLE_OBJECT` just like `DEVICE_DO_SPACES`:

```suggestion
    const DEVICE_ORACLE_OBJECT = 'oracleobject';
```

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Please see my comments and make sure to run the formatter to fix the lint problems.

**Comment by gewenyu99** [comment]
Hey,

Due to time constraints, I'm going to mark this PR `hacktoberfest-accepted` for now so you get DO's Hacktoberfest rewards. We'll continue to work with you on this issue for review and merge.

When it is merged, we'll contact you for Appwrite-specific Hacktoberfest swag.

Thanks for helping us improve Appwrite!

**Comment by stnguyen90** [comment]
Unfortunately, I'm going to need to remove the hacktoberfest-accepted label as what's done so far doesn't actually implement anything.

**Comment by SaketKaswa20** [comment]
No worries! It was great working with you all!

**Comment by iSuslov** [comment]
Well, kind of sad. Are there any plans to support it afterall? Oracle Cloud is undervalued for sure.

**Comment by gewenyu99** [comment]
@iSuslov We'd still love a valid implementation of the Oracle Object Storage adapter. I just think the existing PR wasn't quite ready for merge. 

We could also see a generic S3 adapter in the future, which would also solve the same issue.

TL;DR: We'd love to support it, this PR just wasn't ready for merge :)
